lib/timber/integrations/phoenix_instrumenter.ex

```diff
@@ -72,9 +72,98 @@ defmodule Timber.Integrations.PhoenixInstrumenter do
   require Logger
 
   alias Timber.Event
+  alias Timber.Events.ChannelJoinEvent
+  alias Timber.Events.ChannelReceiveEvent
   alias Timber.Events.ControllerCallEvent
   alias Timber.Events.TemplateRenderEvent
 
+  #
+  # Channels
+  #
+
+  @doc false
+  @spec phoenix_channel_join(:start | :stop, map | non_neg_integer, map | :ok) :: :ok
```

Comment:
I would split this into multiple `spec` statements to specify the exact patterns:

```elixir
@spec phoenix_channel_join(:start, compile_metadata :: map, runtime_metadata :: map) :: :ok
@spec phoenix_channel_join(:stop, time_diff_native :: non_neg_integer, result_of_before_callback :: :ok) :: :ok
```

I've named the arguments inline for the benefit of reader, but they can be removed if you feel them unnecessary.